app/controllers/admin/configs_controller.rb

```diff
@@ -129,10 +129,23 @@ class ConfigsController < Admin::ApplicationController
         home_feed_minimum_score
       ].freeze
 
+    VALID_IMAGE_URLS =
+      %i[
+        logo_png
+        secondary_logo_url
+        mascot_image_url
+        mascot_footer_image_url
+        campaign_sidebar_image
+        onboarding_logo_image
+        onboarding_background_image
+        onboarding_taskcard_image
+      ].freeze
+
     layout "admin"
 
     before_action :extra_authorization_and_confirmation, only: [:create]
     before_action :validate_inputs, only: [:create]
+    before_action :validate_image_urls, only: [:create], if: -> { VALID_IMAGE_URLS.present? }
```

Comment:
This `before_action` is still too aggressive and validates _every_ field regardless of whether or not it is an image on `create` -- any ideas for how to adjust this to only validate images on `create`? 🤔  I've tried a ton of different approaches at this point and still cannot get this to work only for images.

Reply:
So `VALID_IMAGE_URLS.present?` always returns true because the constant is always there. I'm guessing you're trying to do something like:
```
validate_image_urls, only: [:create], if -> { site config params have image fields in them }
```

Since you have `VALID_IMAGE_URLS` already, you want to compare that to the site config params' keys and see if the two arrays have intersecting elements.

Reply:
Great suggestions! I've changed the constant name and will try comparing the constant against the params like you mentioned. 😊 

Thank you so much for the extra set of eyes, @Zhao-Andy!